Sources/SwiftDocC/Infrastructure/Link Resolution/PathHierarchy.swift

```diff
@@ -0,0 +1,1046 @@
+/*
+ This source file is part of the Swift.org open source project
+
+ Copyright (c) 2022 Apple Inc. and the Swift project authors
+ Licensed under Apache License v2.0 with Runtime Library Exception
+
+ See https://swift.org/LICENSE.txt for license information
+ See https://swift.org/CONTRIBUTORS.txt for Swift project authors
+*/
+
+import Foundation
+import SymbolKit
+
+/// An opaque identifier that uniquely identifies a resolved entry in the path hierarchy,
+///
+/// Resolved identifiers cannot be inspected and can only be created by the path hierarchy.
```

Comment:
It would be useful to document the topic identifiers are purposefully being exposed as opaque values.